Run on nightly build of Rust

I have issue while running in rust nightly build.

```
Enter passphrase for key '/home/mauryaempire/.ssh/id_rsa': 
Enter passphrase for key '/home/mauryaempire/.ssh/id_rsa': 
Enter passphrase for key '/home/mauryaempire/.ssh/id_rsa': 
remote: Counting objects: 17, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 17 (delta 5), reused 5 (delta 5), pack-reused 9
Unpacking objects: 100% (17/17), done.
From github.com:rust-lang/rust-by-example
 * branch            master     -> FETCH_HEAD
   3f64903..a1abae9  master     -> upstream/master
Updating 3f64903..a1abae9
Fast-forward
 examples/std_misc/process/pipe/pipe.rs | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)
[mauryaempire@chandragupta rust-by-example]$ make 
./setup-stage.sh
Checked out revision 119.
cargo run
   Compiling regex v0.1.30
   Compiling rustc-serialize v0.3.14
   Compiling update v0.0.1 (file:///home/mauryaempire/Documents/work/projects/rust-by-example)
src/example.rs:40:48: 40:62 error: use of unstable library feature 'iter_arith': bounds recently changed
src/example.rs:40                                               .sum::<usize>(),
                                                                 ^~~~~~~~~~~~~~
src/example.rs:40:48: 40:62 help: add #![feature(iter_arith)] to the crate attributes to enable
src/main.rs:1:12: 1:16 error: unused or unknown feature, #[deny(unused_features)] on by default
src/main.rs:1 #![feature(core)]
                         ^~~~
src/main.rs:29:17: 29:31 error: use of deprecated item: this unsafe API is unlikely to ever be stabilized in this form, #[deny(deprecated)] on by default
src/main.rs:29         let _ = thread::scoped(move || {
                               ^~~~~~~~~~~~~~
note: in expansion of for loop expansion
src/main.rs:25:5: 34:6 note: expansion site
error: aborting due to 3 previous errors
Could not compile `update`.

To learn more, run the command again with --verbose.
Makefile:28: recipe for target 'all' failed
make: *** [all] Error 101
```

--- a/examples/fn/closures/output_parameters/output_parameters.rs
+++ b/examples/fn/closures/output_parameters/output_parameters.rs
@@ -1,4 +1,5 @@
-#![feature(core)]
+#![feature(fnbox)]
+
 use std::boxed::FnBox;
 
 // Return a closure taking no inputs and returning nothing

--- a/examples/fn/hof/hof.rs
+++ b/examples/fn/hof/hof.rs
@@ -1,4 +1,4 @@
-#![feature(core)]
+#![feature(iter_arith)]
 
 fn main() {
     println!("Find the sum of all the squared odd numbers under 1000");

--- a/examples/std_misc/arg/args.rs
+++ b/examples/std_misc/arg/args.rs
@@ -1,4 +1,4 @@
-#![feature(collections)]
+#![feature(slice_extras)]
 
 use std::env;
 

--- a/examples/std_misc/arg/getopts/echo.rs
+++ b/examples/std_misc/arg/getopts/echo.rs
@@ -1,12 +1,12 @@
 #![feature(rustc_private)]
-#![feature(collections)]
-#![feature(exit_status)]
+#![feature(slice_extras)]
 
 extern crate getopts;
 
 use std::env;
 use std::io;
 use std::io::Write;
+use std::process;
 
 static VERSION: &'static str = "1.0.0";
 
@@ -29,8 +29,7 @@ fn main() {
         Ok(m) => m,
         Err(f) => {
             println!("{}", f);
-            env::set_exit_status(1);
-            return;
+            process::exit(1);
             // The exit code is 0 (success) by default.
             // Any exit code other than 0 indicates failure.
         }

--- a/examples/std_misc/arg/getopts/testopt.rs
+++ b/examples/std_misc/arg/getopts/testopt.rs
@@ -1,10 +1,10 @@
 #![feature(rustc_private)]
-#![feature(collections)]
-#![feature(exit_status)]
+#![feature(slice_extras)]
 
 extern crate getopts;
 
 use std::env;
+use std::process;
 
 fn main() {
     let args: Vec<String> = env::args().collect();
@@ -22,15 +22,14 @@ fn main() {
         Ok(m) => m,
         Err(f) => {
             println!("{}", f);
-            env::set_exit_status(1);
-            return;
+            process::exit(1);
         }
     };
     let a = if matches.opt_present("a") {true} else {false};
     let b = if matches.opt_present("b") {true} else {false};
     let c = match matches.opt_str("c") {
         Some(s) => s,
-        None => String::from_str(""),
+        None => String::from(""),
     };
     //^ Use `matches.opt_default` if you need a default (`opflagopt`).
     //  Use `matches.opt_count` if you need to count how many were matched

--- a/examples/std_misc/simd/simd.rs
+++ b/examples/std_misc/simd/simd.rs
@@ -1,4 +1,4 @@
-#![feature(core)]
+#![feature(core_simd)]
 
 use std::simd::f32x4;
 

--- a/examples/std_misc/simd/simd_add.rs
+++ b/examples/std_misc/simd/simd_add.rs
@@ -1,5 +1,5 @@
 #![feature(test)]
-#![feature(core)]
+#![feature(core_simd)]
 
 use std::simd::f32x4;
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -1,7 +1,8 @@
-#![feature(core)]
+#![feature(iter_arith)]
 #![feature(scoped)]
 
 #![deny(warnings)]
+#![allow(deprecated)] // thread::scoped needs to go, but can't do it now
 #![feature(plugin)]
 
 extern crate regex;

